src/Microsoft.TemplateEngine.Edge/Constraints/HostConstraint.cs

```diff
@@ -0,0 +1,204 @@
+                    return TemplateConstraintResult.CreateRestricted(Type, errorMessage);
+                }
+                catch (ConfigurationException ce)
+                {
+                    return TemplateConstraintResult.CreateFailure(Type, ce.Message, LocalizableStrings.Generic_Constraint_WrongConfigurationCTA);
+                }
+            }
+
+            // configuration examples
+            // "args": [
+            //      {
+            //          "hostName": "dotnetcli",
+            //          "version": "5.0.100"
+            //      },
+            //      {
+            //          "hostName": "ide",
+            //          "version": "[16.0-*]"
+            //      }]
+            private static IEnumerable<HostInformation> ParseArgs(string? args)
+            {
+                if (string.IsNullOrWhiteSpace(args))
+                {
+                    throw new ConfigurationException(LocalizableStrings.HostConstraint_Error_ArgumentsNotSpecified);
+                }
+
+                JToken? token;
+                try
+                {
+                    token = JToken.Parse(args!);
+                }
+                catch (Exception e)
+                {
+                    throw new ConfigurationException(string.Format(LocalizableStrings.HostConstraint_Error_InvalidJson, args), e);
+                }
+
+                if (token is not JArray array)
+                {
+                    throw new ConfigurationException(string.Format(LocalizableStrings.HostConstraint_Error_InvalidJsonArray, args));
+                }
+
+                List<HostInformation> hostInformation = new List<HostInformation>();
+
+                foreach (JToken value in array)
+                {
+                    if (value is not JObject jobj)
+                    {
+                        throw new ConfigurationException(string.Format(LocalizableStrings.HostConstraint_Error_InvalidJsonArray_Objects, args));
+                    }
+
+                    string? hostName = jobj.ToString("hostname");
+                    string? version = jobj.ToString("version");
+
+                    if (string.IsNullOrWhiteSpace(hostName))
+                    {
+                        throw new ConfigurationException(string.Format(LocalizableStrings.HostConstraint_Error_MissingMandatoryProperty, jobj, "hostname"));
+                    }
+                    if (string.IsNullOrWhiteSpace(version))
+                    {
+                        hostInformation.Add(new HostInformation(hostName!));
+                        continue;
+                    }
+
+                    //check version in the following order:
+                    //NuGet exact verion
+                    //NuGet floating version
+                    //NuGet version range
+                    //Legacy template engine exact version
+                    //Legacy template engine version range
```

Comment:
Great!
Do we have the comparison logic documented in the feature documentation?

Reply:
This would be done in separate sub-task of https://github.com/dotnet/templating/issues/3107